src/librustc_typeck/check/coercion.rs

```diff
@@ -1348,6 +1348,34 @@ impl<'tcx, 'exprs, E: AsCoercionSite> CoerceMany<'tcx, 'exprs, E> {
                 "...is found to be `{}` here",
```

Comment:
Extract this `if let` and contents to a new method.

Reply:
That seems a bit excessive to me, this function's entire point is to extend the `DiagnosticBuilder` with this info.

Reply:
The point of the method is to return a `DiagnosticBuilder<'_>` for the purpose of reporting about mismatched types. The `if let` contained here is still a well defined sub-goal of achieving the whole picture and works as a unit.